Content.Server/Atmos/EntitySystems/FlammableSystem.cs

```diff
@@ -494,21 +495,35 @@ public override void Update(float frameTime)
 
                     var ev = new GetFireProtectionEvent();
                     // let the thing on fire handle it
-                    RaiseLocalEvent(uid, ref ev);
-                    // and whatever it's wearing
+                    RaiseLocalEvent(uid, ref ev);                    // and whatever it's wearing
                     if (_inventoryQuery.TryComp(uid, out var inv))
                         _inventory.RelayEvent((uid, inv), ref ev);
 
                     DamageSpecifier? damage;
                     if (HasComp<XenoComponent>(uid))
+                    {
                         damage = flammable.Intensity * (flammable.FireStacks / flammable.Duration * 0.2 + 0.8) * ev.Multiplier * flammable.Damage / 2;
+                    }
                     else
+                    {
                         damage = flammable.Intensity / 5f * flammable.Damage;
+                    }
 
                     if (_steppingOnFireQuery.HasComp(uid))
-                        damage *= 2;
-
-                    _damageableSystem.TryChangeDamage(uid, damage, true, false);
+                        damage *= 2;// Check fire immunity for DOT damage
+                    bool shouldDealDamage = true;
+                    var hasImmunity = TryComp<RMCImmuneToFireTileDamageComponent>(uid, out var immunity);
+                    var hasBypassComponent = HasComp<RMCFireBypassActiveComponent>(uid);
+                    if (hasImmunity)
```

Comment:
You can remove this boolean and put the trycomp rmcimmune in there instead, then remove hasBypassComponent and make shouldDealDamage = HasComp